documentation/asciidoc/computers/camera/rpicam_apps_building.adoc

```diff
@@ -1,172 +1,256 @@
 
-Rebuilding `libcamera` from scratch should be necessary only if you need the latest features that may not yet have reached the `apt` repositories, or if you need to customise its behaviour in some way.
+NOTE: Only build `libcamera` from scratch if you need custom behaviour or the latest features that have not yet reached `apt` repositories.
 
-First install all the necessary dependencies for `libcamera`. Raspberry Pi OS Lite users will first need to install the following additional packages if they have not done so previously:
+[NOTE]
+======
+If you run Raspberry Pi OS Lite, begin by installing the following packages:
 
+[source,console]
 ----
-sudo apt install -y python3-pip git python3-jinja2
+$ sudo apt install -y python3-pip git python3-jinja2
 ----
+======
 
-All users should then install the following:
+First, install the following `libcamera` dependencies:
 
+[source,console]
 ----
-sudo apt install -y libboost-dev
-sudo apt install -y libgnutls28-dev openssl libtiff5-dev pybind11-dev
-sudo apt install -y qtbase5-dev libqt5core5a libqt5gui5 libqt5widgets5
-sudo apt install -y meson cmake
-sudo apt install -y python3-yaml python3-ply
+$ sudo apt install -y libboost-dev
+$ sudo apt install -y libgnutls28-dev openssl libtiff5-dev pybind11-dev
+$ sudo apt install -y qtbase5-dev libqt5core5a libqt5gui5 libqt5widgets5
+$ sudo apt install -y meson cmake
+$ sudo apt install -y python3-yaml python3-ply
+$ sudo apt install -y libglib2.0-dev libgstreamer-plugins-base1.0-dev
 ----
 
-In the `meson` commands below we have enabled the gstreamer plugin. If you do not need this, you can set `-Dgstreamer=disabled` instead and the next pair of dependencies will not be required. But if you do leave _gstreamer_ enabled, then you will need the following:
+Now we're ready to build `libcamera` itself.
 
+Download a local copy of Raspberry Pi's fork of `libcamera` from GitHub:
+
+[source,console]
 ----
-sudo apt install -y libglib2.0-dev libgstreamer-plugins-base1.0-dev
+$ git clone https://github.com/raspberrypi/libcamera.git
 ----
 
-Now we can check out and build `libcamera` itself. We check out Raspberry Pi's fork of `libcamera`, which tracks the official repository but lets us control exactly when we pick up new features.
+Navigate into the root directory of the repository:
 
+[source,console]
 ----
-cd
-git clone https://github.com/raspberrypi/libcamera.git
-cd libcamera
+$ cd libcamera
 ----
 
-Now run:
+Next, run `meson` to configure the build environment:
 
+[source,console]
 ----
-meson setup build --buildtype=release -Dpipelines=rpi/vc4,rpi/pisp -Dipas=rpi/vc4,rpi/pisp -Dv4l2=true -Dgstreamer=enabled -Dtest=false -Dlc-compliance=disabled -Dcam=disabled -Dqcam=disabled -Ddocumentation=disabled -Dpycamera=enabled
+$ meson setup build --buildtype=release -Dpipelines=rpi/vc4,rpi/pisp -Dipas=rpi/vc4,rpi/pisp -Dv4l2=true -Dgstreamer=enabled -Dtest=false -Dlc-compliance=disabled -Dcam=disabled -Dqcam=disabled -Ddocumentation=disabled -Dpycamera=enabled
```

Comment:
Yep! We actually do this in a lot of other places in the documentation. It's meant to make it obvious that you're dealing with lines intended to be entered into a CLI.

I would like to enhance our copy/paste snippet workflow to include a button that ignores the symbol. And, better yet, make the $ symbol nonselectable so even old school manual copy/paste can't grab it accidentally. But we haven't done that yet.

Right now, it's there because it's useful to delineate one-liner snippets meant for the CLI (or collections of commands run on the CLI in a single snippet) from commands meant to be written into a file. We haven't always made that distinction obvious in the past. For now, this is as good as it gets, but I will look into improvements.

Reply:
When long command-lines get word-wrapped (which is more of a problem in the PDF documentation than the HTML documentation), it also makes it clearer which lines are "new command lines" and which lines are merely "a continuation of the previous line".

Reply:
IMHO, having the ability to easily copy/paste single or multiple commands from the documentation is a massive quality of life improvement.  I would advocate getting this working asap.

Reply:
I don't disagree, and it's on the list.

Reply:
Added to the list: https://github.com/raspberrypi/documentation/issues/3556